[tvOS 16] Support building WebKit

emw-apple opened the pull request:
#### 1da4d67f46d2eb896fc9182411454756599817b2
<pre>
[tvOS 16] Support building WebKit
<a href="https://bugs.webkit.org/show_bug.cgi?id=245095">https://bugs.webkit.org/show_bug.cgi?id=245095</a>

Reviewed by Jonathan Bedard, Kimmo Kinnunen and Alex Christensen.

* Source/WebCore/PAL/pal/spi/cocoa/MetalSPI.h: The use site for
  MTLSharedTextureHandle is behind a !PLATFORM(IOS_FAMILY_SIMULATOR)
  block, but this SPI declaration was not.

* Source/WebKit/Scripts/webkit/messages.py: Move
  RemoteMediaSourceIdentifier, RemoteSourceBufferIdentifier,
  XRDeviceIdentifier to types_that_cannot_be_forward_declared. These
  symbols are not always available (in this case when building tvOS), so
  they cannot be generated as part of IPC::serializedIdentifiers.

* Source/WebKit/Scripts/webkit/tests/MessageArgumentDescriptions.cpp:
(IPC::serializedIdentifiers): Regenerated.

Canonical link: <a href="https://commits.webkit.org/254459@main">https://commits.webkit.org/254459@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/aa47a650fbf4367f5daccd37aab6f19fa9b0a6ec

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/89016 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/33580 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/19873 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/98308 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/154660 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/93025 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/32085 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/27664 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/81381 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/92825 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/94663 "Passed tests") | [⏳ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/iOS-15-Simulator-WK2-Tests-EWS "Waiting to run tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/75975 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/25409 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/92603 "Passed tests") | [⏳ 🧪 api-ios](https://ews-build.webkit.org/#/builders/API-Tests-iOS-Simulator-EWS "Waiting to run tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/80348 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/68378 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/29874 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/14403 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/29607 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/15379 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3124 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/33046 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/38325 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/31736 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/34473 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

emw-apple commented on Source/WebKit/Scripts/webkit/messages.py:
```diff
@@ -376,13 +373,16 @@ def types_that_cannot_be_forward_declared():
         'WebKit::ImageBufferBackendHandle',
         'WebKit::LayerHostingContextID',
         'WebKit::LegacyCustomProtocolID',
+        'WebKit::RemoteMediaSourceIdentifier',
+        'WebKit::RemoteSourceBufferIdentifier',
         'WebKit::RemoteVideoFrameWriteReference',
         'WebKit::RemoteVideoFrameReadReference',
         'WebKit::RenderingUpdateID',
         'WebKit::TextCheckerRequestID',
         'WebKit::TransactionID',
         'WebKit::WCLayerTreeHostIdentifier',
         'WebKit::WCContentBufferIdentifier',
+        'WebKit::XRDeviceIdentifier',
```
This change is a bit of a stab in the dark and could use review from @achristensen07. The `serialized_identifiers()` generator was added recently in https://github.com/WebKit/WebKit/commit/d32d14b0307fbac280cfb1b55835657d63daef7d and doesn't have a way to conditionally reference these types. I'm not sure why the tvOS 16 SDK would behave differently here, as these symbols should _also_ be missing in the current tvOS EWS build. But this is what I needed to get my build working.

kkinnunen-apple replied:
EWS likely does not build debug / release+asan for tvOS.
The IPC test api feature is enabled only on these.

kkinnunen-apple commented on Source/WebKit/Scripts/webkit/messages.py:
```diff
@@ -330,7 +328,6 @@ def serialized_identifiers():
         'WebKit::WebPageProxyIdentifier',
         'WebKit::WebURLSchemeHandlerIdentifier',
-        'WebKit::XRDeviceIdentifier',
     ]
```
Alex can comment more, but the condition for the identifier could be added to the hardcoded list. E.g.
`('WebKit::XRDeviceIdentifier, 'ENABLE(WEBXR)'),`

and then the generator iteration part can do
```
if isinstance(I, str):
    ....
else # assume tuple (name, condition)
```

emw-apple replied:
Seems straightforward enough. I can implement this if you'd like.

kkinnunen-apple reviewed (approved):
approved with the changes described inline

achristensen07 reviewed (approved):
This is fine for now.  We can add identifiers that have conditions later.

emw-apple committed:
[tvOS 16] Support building WebKit
https://bugs.webkit.org/show_bug.cgi?id=245095

Reviewed by Jonathan Bedard, Kimmo Kinnunen and Alex Christensen.

* Source/WebCore/PAL/pal/spi/cocoa/MetalSPI.h: The use site for
  MTLSharedTextureHandle is behind a !PLATFORM(IOS_FAMILY_SIMULATOR)
  block, but this SPI declaration was not.

* Source/WebKit/Scripts/webkit/messages.py: Move
  RemoteMediaSourceIdentifier, RemoteSourceBufferIdentifier,
  XRDeviceIdentifier to types_that_cannot_be_forward_declared. These
  symbols are not always available (in this case when building tvOS), so
  they cannot be generated as part of IPC::serializedIdentifiers.

* Source/WebKit/Scripts/webkit/tests/MessageArgumentDescriptions.cpp:
(IPC::serializedIdentifiers): Regenerated.

Canonical link: https://commits.webkit.org/254459@main
